torch/_inductor/utils.py

```diff
@@ -231,6 +231,27 @@ def has_incompatible_cudagraph_ops(gm):
     return False
 
 
+def create_static_inputs(inps, known_static_input_idxs):
+    static_inps = []
+    dst_buffers = {}
+    # Create new shared storage for input tensors that have the same underlying storage.
+    # Use existing tensor as-is for input tensors tracked in `known_static_input_idxs`.
+    for idx, x in enumerate(inps):
+        if idx not in known_static_input_idxs:
+            src_storage = x.storage()
+            # If common buffer is not created for this storage, create it; otherwise retrieve it.
+            if src_storage.data_ptr() not in dst_buffers:
+                dst_buffers[src_storage.data_ptr()] = torch.empty(
+                    src_storage.size(), dtype=src_storage.dtype, device=src_storage.device)
+            dst_buffer = dst_buffers[src_storage.data_ptr()]
+            storage_offset = (x.data_ptr() - x.storage().data_ptr()) // x.element_size()
```

Comment:
For Triton we require all inputs to be aligned to 16 bytes. (`x.data_ptr()%16==0`)

The previous version of static inputs had the side effect of also aligning the input, because `torch.zeros()` always returns aligned data.  This new version will preserve unaligned inputs, which will break things if Triton vectorizes a load from that buffer.

There is other code in this file for handling this in the non-cudagraphs case.

Reply:
Thanks - would this work?
```python
new_dst_buffer = torch.zeros(src_storage.size(), dtype=src_storage.dtype, device=src_storage.device)
assert new_dst_buffer.data_ptr() % 16 == 0
```

Reply:
The test case to check would be:
```
x, = create_static_inputs([torch.zeros(128)[1:]], {})
assert x.data_ptr() % 16 == 0
```